awx/main/management/commands/cleanup_host_metrics.py

```diff
@@ -1,22 +1,24 @@
-from awx.main.models import HostMetric
 from django.core.management.base import BaseCommand
-from django.conf import settings
+from django.utils.translation import gettext_lazy as _
+from awx.main.tasks.host_metrics import HostMetricTask
 
 
 class Command(BaseCommand):
     """
-    Run soft-deleting of HostMetrics
+    Run soft/hard-deleting of HostMetrics
     """
 
-    help = 'Run soft-deleting of HostMetrics'
+    help = 'Run soft/hard-deleting of HostMetrics'
```

Comment:
I'm realizing that I'd like a little more more substance behind these docs. After reading the querysets, I will use my own words to say that.

Soft deleting will _mark_ all host metrics as deleted if they are older than the specified number of months. This is the same as when you delete those host metrics via the API, so this is just a way to automatically do that after a certain amount of time.

Hard deleting will fully delete all host metrics that fit 2 criteria - that they are older than the specified number of months AND they have already been soft-deleted. Practically, I can see why the former criteria shouldn't usually matter, because the soft deletion threshold _should_ be shorter than the hard deletion threshold, but I don't see any guarantees of this.

Reply:
Not super familiar with this, and I could be dead wrong, but it looks like if `hardlimit=softlimit`, then behavior would be order-dependent. Like if soft runs before hard vs. hard running before soft: Wouldn't you get different results?

Reply:
The order is that soft deletion runs before hard deletion, so assuming `softlimit<hardlimit` then anything older than the hard limit will be deleted. It appears possible that un-deleted records older than the hard limit will be kept if they have never been soft-deleted.

Reply:
to solve these assumptions, we can 
- make both --soft and --hard arguments required and check if soft < hard
- rely on settings only, so no arguments possible (but then it'll be harder to test it), but it'll do the same like scheduler does

--- 

about the comments, when this review targets to the "help", I don't see any multiline "help" string among all commands. So we can keep one-liner for the help and extend class comment, ok?

Reply:
I don't think we should hesitate to have multi-line command help text. Example:

https://github.com/ansible/awx/blob/devel/awx/main/management/commands/register_default_execution_environments.py

You could, for example, add a sentence to this help that says "this is also run periodically as a system task" to establish more context.

But I don't think you need the details discussed in this thread in the _command_ help text, since they can mostly be covered in the help text for the `--hard` and `--soft` options, which is discussed in another thread.

> check if soft < hard

No objections to this. It's fairly common to see `raise CommandError('you did this wrong')`, and this is often better than documenting the restriction (since users often will not read it).

Reply:
I got an answer that it's not wanted to allow configurable command. It'll be driven by settings only.  Accordingly I'll update the comments